keps/sig-cloud-provider/0003-testgrid-conformance-e2e.md

```diff
@@ -0,0 +1,270 @@
+
+##### Installing Sonobuoy
+
+The following is mostly a [copy of the sonobuoy documentation](https://github.com/heptio/sonobuoy#download-and-run).
+
+First install sonobuoy. The following command adds it to the `$GOBIN` environment variable which is expected to be part of your `$PATH` environment variable.
+```
+$ go get -u github.com/heptio/sonobuoy
+```
+
+##### Running Conformance Tests with Sonobuoy
+
+You can then start e2e tests by simply running:
+```
+$ sonobuoy run
+Running plugins: e2e, systemd-logs
+INFO[0001] created object                                name=heptio-sonobuoy namespace= resource=namespaces
+INFO[0001] created object                                name=sonobuoy-serviceaccount namespace=heptio-sonobuoy resource=serviceaccounts
+INFO[0001] created object                                name=sonobuoy-serviceaccount-heptio-sonobuoy namespace= resource=clusterrolebindings
+INFO[0001] created object                                name=sonobuoy-serviceaccount namespace= resource=clusterroles
+INFO[0001] created object                                name=sonobuoy-config-cm namespace=heptio-sonobuoy resource=configmaps
+INFO[0002] created object                                name=sonobuoy-plugins-cm namespace=heptio-sonobuoy resource=configmaps
+INFO[0002] created object                                name=sonobuoy namespace=heptio-sonobuoy resource=pods
+INFO[0002] created object                                name=sonobuoy-master namespace=heptio-sonobuoy resource=services
+```
+
+You can then check the status of your e2e tests by running
+```
+$ sonobuoy status
+PLUGIN		STATUS	COUNT
+e2e		running	1
+systemd_logs	running	3
+
+Sonobuoy is still running. Runs can take up to 60 minutes.
+```
+
+E2E tests can take up to an hour. Once your tests are done, you can download a snapshot of your results like so:
+```
+$ sonobuoy retrieve .
+```
+
+Once you have the following snapshot, extract it's contents like so
+```
+$ mkdir ./results; tar xzf *.tar.gz -C ./results
+```
+
+At this point you should have the log file and JUnit results from your tests:
+```
+results/plugins/e2e/results/e2e.log
+results/plugins/e2e/results/junit_01.xml
+```
+
+#### Kubetest
+
+You should use kubetest if you want to run the latest set of tests in upstream Kubernetes. This is highly recommended by sig-testing so that new tests can be accounted for in new releases.
+
+##### Installing Kubetest
+
+Install kubetest using the following command which adds it to your `$GOBIN` environment variable which is expected to be part of your `$PATH` environment variable.
+```
+go get -u k8s.io/test-infra/kubetest
+```
+
+##### Running Conformance Test with kubetest
+
+Now you can run conformance test with the following:
+```
+cd /path/to/k8s.io/kubernetes
+export KUBERNETES_CONFORMANCE_TEST=y
+kubetest \
+    # conformance tests aren't supposed to be aware of providers
+    --provider=skeleton \
```

Comment:
Is `skeleton` what we want?  Or is my previous comment incorrect re: the goal is testing cloud provider specific implementations?  I should take a closer look at what tests are actually run, but my hunch is we might want a custom set of tests vs conformance for each cloud provider to actually catch issues.